web/projects/ui/src/app/services/state.service.ts

```diff
@@ -0,0 +1,116 @@
+import { inject, Injectable } from '@angular/core'
+import { CanActivateFn, IsActiveMatchOptions, Router } from '@angular/router'
+import { ALWAYS_TRUE_HANDLER } from '@taiga-ui/cdk'
+import { TuiAlertService, TuiNotification } from '@taiga-ui/core'
+import {
+  BehaviorSubject,
+  combineLatest,
+  concat,
+  exhaustMap,
+  from,
+  merge,
+  Observable,
+  retry,
+  startWith,
+} from 'rxjs'
+import {
+  filter,
+  map,
+  shareReplay,
+  skip,
+  switchMap,
+  takeUntil,
+  tap,
+} from 'rxjs/operators'
+import { RR } from 'src/app/services/api/api.types'
+import { ApiService } from 'src/app/services/api/embassy-api.service'
+import { ConnectionService } from 'src/app/services/connection.service'
+
+const OPTIONS: IsActiveMatchOptions = {
+  paths: 'subset',
+  queryParams: 'exact',
+  fragment: 'ignored',
+  matrixParams: 'ignored',
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class StateService extends Observable<RR.ServerState | null> {
+  private readonly alerts = inject(TuiAlertService)
+  private readonly api = inject(ApiService)
+  private readonly router = inject(Router)
+  private readonly connection = inject(ConnectionService)
+  private readonly trigger$ = new BehaviorSubject<void>(undefined)
+  private readonly stream$ = this.trigger$.pipe(
+    switchMap(() =>
+      from(this.api.getState()).pipe(retry({ delay: 2000 }), startWith(null)),
+    ),
+    tap(state => {
```

Comment:
If null is necessary, shouldn't there be a case for it in the switch statement? At least a default case?

Reply:
No, it doesn't do anything, it means we don't know the state and have no place to redirect you yet, so it is just ignored.